src/pynwb/file.py

```diff
@@ -837,7 +816,12 @@ def _check_sweep_table(self):
         Create a SweepTable if not yet done.
         """
         if self.sweep_table is None:
-            self.sweep_table = SweepTable(name='sweep_table')
+            if self._in_construct_mode:
+                sweep_table = SweepTable.__new__(SweepTable, parent=self, in_construct_mode=True)
+                sweep_table.__init__(name='sweep_table')
```

Comment:
```suggestion
            if self._in_construct_mode:
                # Construct the SweepTable without triggering errors in construct mode because SweepTable has been deprecated
                sweep_table = SweepTable.__new__(SweepTable, parent=self, in_construct_mode=True)
                sweep_table.__init__(name='sweep_table')
                sweep_table._in_construct_mode = False
```